js/src/volume.js

```diff
@@ -279,6 +282,20 @@ var ScatterView = widgets.WidgetView.extend( {
         _.mapObject(this.model.changedAttributes(), function(val, key){
             console.log("changed " +key)
             this.previous_values[key] = this.model.previous(key)
+            if (key == "index"){
+              pindex = this.previous_values["index"]
+              this.previous_values["x"] = this.model.previous("x")[pindex]
+              this.previous_values["y"] = this.model.previous("y")[pindex]
+              this.previous_values["z"] = this.model.previous("z")[pindex]
+              var vx = this.model.previous("vx")
+              if (vx && typeof vx[0][0] != "undefined" ) {
+                this.previous_values["vx"] = this.model.previous("vx")[pindex]
+                this.previous_values["vy"] = this.model.previous("vy")[pindex]
+                this.previous_values["vz"] = this.model.previous("vz")[pindex]
+              }
+
```

Comment:
To make the animation work, for every property that is a sequence (x, y, z, ...)
this.attributes_changed['x'] = ['x']
This tells that 'x' (the first one) is changed, and that after it has changed, it should remove the 'x' from the list previous_values list. Do the same for y, z, vx, etc..
(basically the previous snippet in a look for 'x', 'y', ...